frontend/components/Button/button.cy.tsx

```diff
@@ -0,0 +1,285 @@
+import 'material-symbols';
+import '../../styles/global.css';
+import { Button } from './Button';
+
+describe('Button', () => {
+  describe('Primary Button', () => {
+    it('should display primary button', () => {
+      cy.mount(<Button kind="primary" onClick={() => ""}>Test</Button>)
+      cy.get('button').should('have.class', 'bg-blue-500')
+    })
+
+    describe('Icon', () => {
+      it('should display icon', () => {
+        cy.mount(<Button kind="primary" icon="face" onClick={() => ""}>Test</Button>)
+        cy.get('button i').should('be.visible')
+      })
+    })
+
+    describe('Loading', () => {
+      beforeEach(() => {
+        cy.mount(<Button kind="primary" loading icon="face" onClick={() => ""}>Test</Button>)
+      })
+
+      it('should disable button when loading', () => {
+        cy.get('button').should('have.class', 'text-opacity-75')
```

Comment:
You test against visuals. I would not do this.

https://sqa.stackexchange.com/questions/34319/how-to-check-a-button-is-disabled-using-cypress

Does this work? or with attribute?